From 68ab94e167a6a7bc908c264a8a55e47829b15f6a Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Thu, 20 Oct 2022 07:33:26 +0300
Subject: [PATCH] [SW-108178] gaudi3: fix nrtr lbw rr configuration

PCIE LBW transactions via PIF goes through NRTR1_2CH. Hence, ROM needs
to configure associated RR, i.e mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW to
open or close LBW access to PCIE. This was wrongly configured previously
and this patch fixes the issue.

Change-Id: I55aa62d27a529777921a8512f3fd93c6e7385483
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 0ea38eb..4dcdf82 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -179,12 +179,13 @@
 			rr_ctx->rr_idx);
 
 	regmap_writel(rr_ctx->data_min,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_MIN_0 + rr_dist);
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MIN_0 + rr_dist);
 	regmap_writel(rr_ctx->data_max,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_MAX_0 + rr_dist);
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MAX_0 + rr_dist);
 	regmap_writel(rr_ctx->en_val,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_EN_0 + rr_dist);
-	security_rr_update_hit_reg(mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_HIT_AW,
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_EN_0 + rr_dist);
+
+	security_rr_update_hit_reg(mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_HIT_AW,
 				rr_ctx->rr_idx);
 }
 
@@ -213,13 +214,13 @@
 			rr_ctx->rr_idx);
 
 	regmap_writel(rr_ctx->data_min,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_MIN_SHORT_0 + rr_dist);
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MIN_SHORT_0 + rr_dist);
 	regmap_writel(rr_ctx->data_max,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_MAX_SHORT_0 + rr_dist);
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MAX_SHORT_0 + rr_dist);
 	regmap_writel(rr_ctx->en_val,
-			mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_RANGE_EN_SHORT_0 + rr_dist);
+			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_EN_SHORT_0 + rr_dist);
 
-	security_rr_update_hit_reg(mmD0_NRTR0_3CH_CTRL0_RR_LBW_PRIV_HIT_AW,
+	security_rr_update_hit_reg(mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_HIT_AW,
 				rr_ctx->rr_idx);
 }
 
